src/rt/cover.c

```diff
@@ -715,6 +794,77 @@ void cover_count_tags(cover_tagging_t *tagging, int32_t *n_stmts,
    }
 }
 
+void cover_load_spec_file(cover_tagging_t *tagging, const char *path)
+{
+   cover_exclude_ctx_t ctx = {};
+
+   ctx.ef = fopen(path, "r");
+   if (ctx.ef == NULL)
+      fatal_errno("failed to open coverage specification file: %s\n", path);
+
+   tagging->spec = xcalloc(sizeof(cover_spec_t));
+
+   char *delim = " ";
+   ssize_t read;
+   loc_file_ref_t file_ref = loc_file_ref(path, NULL);
+   int line_num = 1;
+   size_t line_len;
+
+   while ((read = getline(&(ctx.line), &line_len, ctx.ef)) != -1) {
+
+      // Strip comments and newline
+      char *p = ctx.line;
+      for (; *p != '\0' && *p != '#' && *p != '\n'; p++);
+      *p = '\0';
+
+      ctx.loc = get_loc(line_num, 0, line_num, p - ctx.line - 1, file_ref);
+
+      // Parse line as space separated tokens
+      for (char *tok = strtok(ctx.line, delim); tok; tok = strtok(NULL, delim)) {
+         bool include;
+         if (tok[0] == '+')
+            include = true;
+         else if (tok[0] == '-')
+            include = false;
+         else
+            fatal_at(&ctx.loc, "Coverage specification command must "
+                               "start with '+' or '-");
+         tok++;
+
+         tb_array_t *arr;
+         if (!strcmp(tok, "hierarchy")) {
+            if (include)
+               arr = &(tagging->spec->hier_include);
+            else
+               arr = &(tagging->spec->hier_exclude);
+         }
+         else if (!strcmp(tok, "block")) {
+            if (include)
+               arr = &(tagging->spec->block_include);
+            else
+               arr = &(tagging->spec->block_exclude);
+         }
+         else
+            fatal_at(&ctx.loc, "invalid command: $bold$%s$$", tok);
+
+         char *hier = strtok(NULL, delim);
+         if (!hier)
+            fatal_at(&ctx.loc, "%s name missing", tok);
+
+         text_buf_t *tb = tb_new();
+         tb_printf(tb, "%s", hier);
+         tb_upcase(tb);
+         APUSH(*arr, tb);
```

Comment:
```suggestion
         APUSH(*arr, xstrdup(hier));
```

See the comment above about using `char *`.